src/radical/pilot/agent/lm/jsrun.py

```diff
@@ -110,18 +110,19 @@ def construct_command(self, cu, launch_script_hop):
         task_argstr  = self._create_arg_string(task_args)
         task_sandbox = ru.Url(cu['unit_sandbox']).path
 
-        self._log.debug('prep %s', cu['uid'])
+        assert(slots), 'missing slots for %s' % uid
```

Comment:
May I suggest something like:
```suggestion
        if not slots:
            raise raise AssertionError( 'missing slots for %s' % uid)
```
As far as I know, asserts can be closed completely, so it might make sense to test data existence and raise exceptions.

Reply:
Nah, then I rather take out the assert.  I don't want to start testing regularly for conditions which are only interesting when debugging.  That assertions can be disabled globally ( via `python -O` or `export PYTHONOPTIMIZE=1`) is a good thing, and exactly why I like to use them - they should not be part of the normal flow of operation.  Thoughts?

Reply:
ping